src/operator/nn/mkldnn/mkldnn_deconvolution.cc

```diff
@@ -83,14 +83,12 @@ std::shared_ptr<deconv_fwd_pd_t> MKLDNNDeconvFwd::CreatePrimitiveDesc(
   const auto get_out_size     = [&pd]() { return pd->dst_desc().get_size(); };
 
   while (!ddc.CheckImplSizeReq(get_data_size(), get_weights_size(), get_out_size())) {
```

Comment:
while -> if?

Reply:
No, ImposePlainWherePadding imposes a plain format on one descriptor at a time: first the data descriptor, then the output descriptor, and the weight descriptor at the end. This way, maybe a better implementation will be selected, than the fully plain one.

Reply:
But I could change it to use `if` and impose a plain format on all descriptors at once

Reply:
If you are certain of a better implementation actually being chosen after imposing plain format to just one/two tensors then it should stay as is, but if not I believe it would be better to have them all flattened at once.

Reply:
Ok, lets keep it this way.